kernel/signal.c

```diff
@@ -488,13 +552,41 @@ static long _handle_one_signal(
         /* save the original fsbase */
         void* original_fsbase = myst_get_fsbase();
         stack_t* altstack = &thread->signal.altstack;
+        uint64_t rsp_before_signal = 0;
 
         /* restore the user-space fsbase, which is pthread_self() */
         myst_set_fsbase(thread->crt_td);
 
+        /* get the rsp value used to determine whether the context before
+         * signal was on the alternative stack
+         * a. for the case of real-time exception (mcontext is not NULL),
```

Comment:
I guess mcontext would also be populated for hardware exceptions like SIGILL, SIGFPE, and SIGSEGV?

Reply:
By real-time I mean all the non-delayed exceptions. Will update the comment.